custom_components/tuya_local/devices/create_warm_clear_heater.yaml

```diff
@@ -0,0 +1,159 @@
+name: Vertical Heater
+products:
+  - id: ggz9d3bputnvntlx
+    manufacturer: Create
+    model: Warm Clear
+entities:
+  - entity: climate
+    translation_only_key: heater
+    dps:
+      - id: 1
+        type: boolean
+        name: hvac_mode
+        mapping:
+          - dps_val: false
+            value: "off"
+          - dps_val: true
+            value: heat
+      - id: 2
+        type: integer
+        name: temperature
+        range:
+          min: 5
+          max: 35
+        unit: C
+      - id: 3
+        type: integer
+        name: current_temperature
+      - id: 4
+        type: string
+        name: preset_mode
+        mapping:
+          - dps_val: CF
+            value: comfort
+          - dps_val: EC
+            value: eco
+          - dps_val: AF
+            value: away
+          - dps_val: P1
+            value: program1
+          - dps_val: P2
+            value: progran2
+          - dps_val: P3
+            value: program3
+          - dps_val: ST
+            value: scheduled
+      - id: 11
+        type: string
+        name: hvac_action
+        mapping:
+          - dps_val: standby
+            value: idle
+          - dps_val: working
+            value: heating
+      - id: 26
+        type: text
```

Comment:
Type should be string, base64 or hex (usually if it is documented as "raw" in Tuya cloud metadata, it is base64, otherwise if it is documented as text or string, then it can be arbitrary readable string data, or hex encoded binary and you need to judge by observing the data, if it is documented as "enum" then there should also be a fixed set of values documented). There are also cases where utf16b64 would be appropriate, but for a "custom_program" I expect it will be binary data encoded in either hex or base64.

Reply:
In this context, there won't be any real difference between string, hex and base64, since the latter 2 only get decoded if there is further processing (like mask or format tags) but it might help future improvements if the type is tagged correctly.